scarb/src/compiler/db.rs

```diff
@@ -93,16 +93,23 @@ fn build_project_config(unit: &CompilationUnit) -> Result<ProjectConfig> {
         .components
         .iter()
         .map(|component| {
+            let experiments = component
+                .package
+                .manifest
+                .experiments
+                .clone()
+                .unwrap_or_default();
             (
                 component.cairo_package_name(),
                 CrateSettings {
                     edition: component.package.manifest.edition,
-                    experimental_features: Default::default(),
+                    experimental_features: cairo_lang_filesystem::db::ExperimentalFeaturesConfig {
+                        negative_impls: experiments.contains("negative_impls"),
+                    },
```

Comment:
I have an idea how to ensure in compile-time that the list of experiments is exhaustive:

Write a macro ([`macro_rules!`](https://blog.logrocket.com/macros-in-rust-a-tutorial-with-examples/)), which usage would look like this:

```rust
experimental_features!(ExperimentalFeatues, [negative_impls,something_else,etc])
```

which would expand to something like this:

```rust
// Struct name is the first argument to the macro.
struct ExperimentalFeatures(HashSet<SmolStr>);

impl ExperimentalFeatures {
    fn new_checked(values: HashSet<SmolStr>, package: PackageId, ui: &Ui) -> Self {
        for feature in &values {
             match feature {
                 "negative_impls" => {},
                 "something_else" => {},
                 "etc" => {},
                 _ => ui.warn(&format!("{package}: unknown experimental feature: {feature}"))
             }
        }
        Self(values)
    }

    fn to_compiler_features(&self) -> cairo_lang_filesystem::db::ExperimentalFeaturesConfig {
        // Rust compiler will check exhaustiveness here
        cairo_lang_filesystem::db::ExperimentalFeaturesConfig {
            negative_impls: self.0.contains("negative_impls"),
        }
    }
}
```

---

this can be done in separate PR; please create an issue for this

Reply:
Issue #1040
This should be also done in `snforge-test-collector` [here](https://github.com/software-mansion/scarb/blob/8733f15164d767a4581a0be1835bd3bd0516244d/extensions/scarb-snforge-test-collector/src/metadata.rs#L92), right?

Reply:
yes, though I don't like this fact